lib/usage-tracking/class-usage-tracking.php

```diff
@@ -0,0 +1,454 @@
+		update_option( $this->hide_tracking_opt_in_option_name, true );
+	}
+
+	/**
+	 * Determine whether the opt-in for enabling usage tracking is hidden.
+	 *
+	 * @access private
+	 *
+	 * @return bool true if the opt-in is hidden, false otherwise.
+	 **/
+	private function _is_opt_in_hidden() {
+		return (bool) get_option( $this->hide_tracking_opt_in_option_name );
+	}
+
+	/**
+	 * If needed, display opt-in dialog to enable tracking.
+	 *
+	 * @access private
+	 **/
+	function _maybe_display_tracking_opt_in() {
+		$opt_in_hidden = $this->_is_opt_in_hidden();
+		$user_tracking_enabled = $this->is_tracking_enabled();
+		$can_manage_tracking = $this->current_user_can_manage_tracking();
+
+		if ( ! $user_tracking_enabled && ! $opt_in_hidden && $can_manage_tracking ) { ?>
+			<div id="<?php echo self::PREFIX; ?>-usage-tracking-notice" class="notice notice-info"
+				data-nonce="<?php echo wp_create_nonce( 'tracking-opt-in' ) ?>">
+				<p>
+					<?php echo $this->opt_in_dialog_text(); ?>
+				</p>
+				<p>
+					<button class="button button-primary" data-enable-tracking="yes">
+						<?php _e( 'Enable Usage Tracking', 'wp-plugin-usage-tracking' ) ?>
+					</button>
+					<button class="button" data-enable-tracking="no">
+						<?php _e( 'Disable Usage Tracking', 'wp-plugin-usage-tracking' ) ?>
+					</button>
+					<span id="progress" class="spinner alignleft"></span>
+				</p>
+			</div>
+			<div id="<?php echo self::PREFIX; ?>-usage-tracking-enable-success" class="notice notice-success hidden">
+				<p><?php _e( 'Usage data enabled. Thank you!', 'wp-plugin-usage-tracking' ) ?></p>
+			</div>
+			<div id="<?php echo self::PREFIX; ?>-usage-tracking-disable-success" class="notice notice-success hidden">
+				<p><?php _e( 'Disabled usage tracking.', 'wp-plugin-usage-tracking' ) ?></p>
+			</div>
+			<div id="<?php echo self::PREFIX; ?>-usage-tracking-failure" class="notice notice-error hidden">
+				<p><?php _e( 'Something went wrong. Please try again later.', 'wp-plugin-usage-tracking' ) ?></p>
+			</div>
+		<?php
+		}
+	}
+
+	/**
+	 * Handle ajax request from the opt-in dialog.
+	 *
+	 * @access private
+	 **/
+	function _handle_tracking_opt_in() {
+		check_ajax_referer( 'tracking-opt-in', 'nonce' );
+
+		if ( ! $this->current_user_can_manage_tracking() ) {
+			wp_die( '', '', 403 );
+		}
+
+		$enable_tracking = isset( $_POST['enable_tracking'] ) && $_POST['enable_tracking'] === '1';
+		$this->set_tracking_enabled( $enable_tracking );
+		$this->_hide_tracking_opt_in();
+		wp_die();
+	}
+
+	/**
+	 * Ensure that jQuery has been enqueued since the opt-in dialog JS depends
+	 * on it.
+	 *
+	 * @access private
+	 **/
+	function _enqueue_script_deps() {
+		// Ensure jQuery is loaded
+		wp_enqueue_script( self::PREFIX . '_usage-tracking-notice', '',
+			array( 'jquery' ), null, true );
+	}
+
+	/**
+	 * Output the JS code to handle the opt-in dialog.
+	 *
+	 * @access private
+	 **/
+	function _output_opt_in_js() {
+?>
+<script type="text/javascript">
```

Comment:
There is also the `wp_add_inline_script()` for queueing inline JS. Doesn't add much, though ;)